### PR TITLE
Release/3.2.0

### PR DESCRIPTION
**Pull request by paul-gene**
### Added
- 3DS 2 now supported
- Basic validation to Dynamic Descriptor configuration options to alleviate errors in checkout

### Fixed
- PayPal breaking Grouped Product pages
- Return type stopping Swagger from compiling (https://github.com/Thundar)
- Handling of exceptions in GatewayCommand class that would show "blank" errors on checkout
- Broken CSS selector
- Giftcards not working with PayPal
- Reverted a change introduced in 3.1.2 where card details were only stored in the database if 
the Vault config option was enabled. This is because partial invoicing, refunds etc need the stored card data. However,
a bug in core Magento 2.3.1 means that if the Vault is turned off, cards are always shown in customer accounts

### Removed
- Layout options for PayPal buttons, due to the buttons now being rendered separately

### CODE REVIEW

File: view/adminhtml/templates/payment/script.phtml [hunk 1]
```diff
@@ -18,7 +18,7 @@ $code = $block->escapeHtml($block->getCode());
             'jquery',
             'domReady!'
         ], function(Braintree, $) {
-            let config = <?= /* @noEscape */ $block->getPaymentConfig() ?>,
+            let config = <?= $block->escapeJs($block->getPaymentConfig()) ?>,
```
Comment (MSkelton-LDG) [inline view/adminhtml/templates/payment/script.phtml]: escapeJs will escape the quotes - escapeJsQuote() should work here

Comment (paul-gene) [reply view/adminhtml/templates/payment/script.phtml]: `escapeJsQuote` escapes quotes, however, it is marked as deprecated.

Comment (MSkelton-LDG) [reply view/adminhtml/templates/payment/script.phtml]: In that case should any function be used here:

https://magento.stackexchange.com/questions/156368/magento-2-template-security-which-method-to-use
image

As this is handling the output from getPaymentConfig() which returns a json_encode()?

Comment (paul-gene) [reply view/adminhtml/templates/payment/script.phtml]: The escape method was mistakenly used as a code tidy pass was done. It was reverted back to being output without an escape method, as it was a JSON string being output. Crisis averted :)

**Pull request merged**